app/collector/data_collection_engine.py

```diff
@@ -0,0 +1,193 @@
+
+import logging
+from datetime import datetime
+import json
+from app.collector.utils.request_utils import (
+    post_request,
+    patch_request,
+    get_request
+)
+from app.collector.utils.cron_utils import (
+    check_if_cron_is_today
+)
+
+
+import app.collector.utils.constants_utils as constants
+from app.collector.collector import (Collector, GenericTable)
+from app.collector.collector_factory import CollectorFactory
+from app.collector.data_collection_logging import DataCollectionLogging
+from app.collector.data_collection_diff_checker import DataCollectionDiffChecker
+
+import os
+
+class DataCollectionEngine():
+# Class to implement the collection data engine.
+
+    def __init__(self):
+        self.log = DataCollectionLogging()
+        self.diff = DataCollectionDiffChecker(self.log)
+
+    def _format_fully_qualified_name (self, list_values):
+    # Format the fully qualified name.
+        fully_qualified_name = list_values
+        fully_qualified_name = [f.lower().replace(" ", "_") for f in fully_qualified_name]
+        return "-".join(fully_qualified_name)+"99"
+
+    def _process_object (self, route, f_q_name, new_dict):   
+    # Process an object generically using the Limoeiro API.
+        f_q_route = route+"/"+constants.F_Q_NAME_ROUTE
+        # Get the object by the fully qualified name.
+        response_code, current_dict = get_request (f_q_route, f_q_name)
+
+        if response_code == 404:
+            # If the object does not exist, create it using a post request.
+            return post_request (route, new_dict)
+        else:
+            # If the object already exists, update it using a patch request.
+            result_dict = self.diff.diff_objects (new_dict, current_dict, route)
+            return patch_request (route, result_dict["id"], result_dict)
+
+    def _process_database (self, database_name, domain, layer, provider):
+    # Process the object database.
+        f_q_name = [domain['name'], layer['name'], provider['name'], database_name]
```

Comment:
Domain and Layer may be None. I think fqn must be provider + database + schema (optional) + table, separated by dots.

Reply:
And some providers, such as MySQL, doesn't have schema. Maybe the fqn generation could be a function implemented by each class associated to each technology.